### ISSUE TITLE
Accessibility and mark-up: make sure labels are clickable in form elements

### ISSUE THREAD

**Issue by mietcls** [ugent-library/biblio-backoffice#1237]
# Bug description
Accessibility and mark-up: make sure labels are clickable in form elements

# Expected result

1. When I click a form label
2. It focuses on the form element
  - [ ] text area
  - [ ] text input
  - [ ] select box
  - [ ] radio button
  - [ ] checkbox

It works for all form elements in datasets, publications, and on all tabs
- [ ] Description
- [ ] Full text & files
- [ ] People & affiliations
- [ ] Datasets
- [ ] Biblio Messages

**Comment by nicolasfranck** [comment]
Reason: attribute `for` in our labels point to the `name` instead of to the `id`. And we often do not assign an `id` to an input field

**Comment by verheyenkoen** [comment]
FYI: focusability can also work without `for`/`id` attributes if you wrap the `input`/`textarea`/`select` inside the `<label>` element but that is probably not feasible in a two column layout.

**Comment by verheyenkoen** [comment]
Already created branch [fix_issue_1237](https://github.com/ugent-library/biblio-backoffice/tree/fix_issue_1237) for this.

**Comment by verheyenkoen** [comment]
During writing of the test, I found that this will probably not work for the "Keywords" field as the tagify component does not support this in any way. This doesn't work with a regular `<input>` field but with a `<span contenteditable>`. You can set focus programmatically however but that needs separate implementation so probably not worth bothering. I'll prepare assertions for this field but comment them out by default.

**Comment by verheyenkoen** [comment]
Written quite a few tests for all publication type forms and new dataset hitting every field that I could find. Are there any other forms that should be addressed?

To run the tests:  
```sh
$ npm install
$ npm test -- --spec=cypress/e2e/issues/issue-1237.cy.ts
```

**Comment by mietcls** [comment]
@verheyenkoen There are no other forms to be added. Thanks!

**Comment by nicolasfranck** [comment]
Cannot work also on repeatable input elements (what element should you focus on?).

e.g. "language", "isbn" ...

**Comment by nicolasfranck** [comment]
And the access levels on the file edit form already have a label for each checkbox.

**Comment by verheyenkoen** [comment]
Checkbox fields indeed seem to be ok already.

**Comment by verheyenkoen** [comment]
Some extra test scenario's:

- [ ] In the "Full text & Files" tab > "Document details" popup you can select Access level "Embargoed access" and 3 new fields appear: 
  - [ ] Access level during embargo
  - [ ] Access level after embargo
  - [ ] Embargo end
- [ ] For datasets, if you select License "The license is not listed here", an extra field "Other license" appears
